ethereum/ethevents/events.go

```diff
@@ -172,155 +161,81 @@ func (ev *EthereumEvents) AddEventHandler(handler EventHandler) {
 // Events are Queued for 60 seconds before processed in order to avoid possible blockchain
 // reversions.
 // Blocking function (use go routine).
-func (ev *EthereumEvents) SubscribeEthereumEventLogs(ctx context.Context, fromBlock int64) {
-	var sub eth.Subscription
-	var err error
+func (ev *EthereumEvents) SubscribeEthereumEventLogs(ctx context.Context) {
+
 	// Get current block
 	tctx, cancel := context.WithTimeout(ctx, types.EthereumReadTimeout)
 	defer cancel()
-	var lastBlock int64
-	var blk *ethtypes.Block
-	errors := 0
-	for {
-		if errors > 5 {
-			log.Fatal("the web3 client connection is not working")
-		}
-		blk, err = ev.VotingHandle.EthereumClient.BlockByNumber(tctx, nil)
-		if err != nil {
-			log.Errorf("cannot get ethereum block: %s", err)
-			errors++
-			time.Sleep(time.Second * 2)
-			if err := ev.VotingHandle.Connect(ev.DialAddrs); err != nil {
-				log.Warn(err)
-			}
-			continue
-		}
-		lastBlock = blk.Number().Int64()
-		break
+	var err error
+	uLastBlockNumber, err := ev.VotingHandle.EthereumClient.BlockNumber(tctx)
```

Comment:
you only need to do this remote ethereum request if fromBlock==0, right? I'd move it inside the "if".